File: Part 2 - Regression/Section 4 - Simple Linear Regression/myfiles/simple_linear_regression.py
#!/usr/bin/env python2
# -*- coding: utf-8 -*-
"""
Created on Wed May  9 10:53:11 2018

@author: dlog
"""

import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

#####################################################################
# import and split data
#####################################################################
dataset = pd.read_csv('../Salary_Data.csv')
X = dataset.iloc[:,:-1].values  # independent variable, years of experience
Y = dataset.iloc[:,1].values    # dependent variable, salary


#####################################################################
# Splitting data to training and test sets
#####################################################################
from sklearn.cross_validation import train_test_split
X_train, X_test, Y_train, Y_test = train_test_split(X,Y, test_size = 1.0/3.0, random_state = 0)


#####################################################################
# fit simple linear regression model
#####################################################################
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train, Y_train)


#####################################################################
# predict test set results
#####################################################################
Y_pred = regressor.predict(X_test) # vector of predictions of dependent variable, predicted salaries for all observations


#####################################################################
# plotting
#####################################################################

# predictor performance on training set
plt.scatter(X_train, Y_train, color = 'green')
plt.plot(X_train, regressor.predict(X_train), color = 'orange')
plt.title('Salary versus Experience (training set)')
plt.xlabel('Years of Experience')
plt.ylabel('Salary')
#plt.show()
plt.savefig('Salary versus Experience, training set.png', bbox_inches='tight')
plt.close()

# predictor performance on test set
plt.scatter(X_test, Y_test, color = 'blue')
plt.plot(X_train, regressor.predict(X_train), color = 'orange')
plt.title('Salary versus Experience (test set)')
plt.xlabel('Years of Experience')
plt.ylabel('Salary')
#plt.show()
plt.savefig('Salary versus Experience, test set.png', bbox_inches='tight')

File: Part 2 - Regression/Section 4 - Simple Linear Regression/myfiles/simple_linear_regression.R
# author: dlog
# created: 9 May 2018


#####################################################################
# import csv located one directory below
#####################################################################
dataset = read.csv('../Salary_Data.csv')


#####################################################################
# Splitting data to training and test sets
#####################################################################
library(caTools)
set.seed(123)
# split by dependent variable 'Salary'
split = sample.split(dataset$Salary, SplitRatio = 2/3) # output interpretation: TRUE = training set, FALSE = test set
training_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)


#####################################################################
# fit simple linear regression model
#####################################################################
regressor = lm(formula = Salary ~ YearsExperience, data = training_set)
summary(regressor)


#####################################################################
# predict test set results
#####################################################################
Y_pred = predict(regressor, newdata = test_set)

#####################################################################
# plotting
#####################################################################
library(ggplot2)
# predictor performance on training set
ggplot() +
    geom_point(aes(x=training_set$YearsExperience, y=training_set$Salary),
               colour = 'green') +
    geom_line(aes(x=training_set$YearsExperience, y=predict(regressor, newdata = training_set)), 
              colour = 'orange') +
    ggtitle(('Salary versus Experience \ntraining set')) +
    xlab('Years of Experience') +
    ylab('Salary')
ggsave('Salary versus Experience, training set.jpg', width = 6, height = 4, units = "cm", scale = 2, dpi = 130)

# predictor performance on test set
ggplot() +
    geom_point(aes(x=test_set$YearsExperience, y=test_set$Salary),
               colour = 'green') +
    geom_line(aes(x=training_set$YearsExperience, y=predict(regressor, newdata = training_set)), 
              colour = 'orange') +
    ggtitle('Salary versus Experience \ntest set') +
    xlab('Years of Experience') +
    ylab('Salary')
ggsave('Salary versus Experience, test set.jpg', width = 6, height = 4, units = "cm", scale = 2, dpi = 130)

